tmail-backend/jmap/extensions-rabbitmq/src/main/java/com/linagora/tmail/james/jmap/RabbitMQEmailAddressContactConfiguration.java

```diff
@@ -0,0 +1,94 @@
+package com.linagora.tmail.james.jmap;
+
+import java.net.URI;
+
+import org.apache.commons.configuration2.Configuration;
+import org.apache.james.backends.rabbitmq.RabbitMQConfiguration;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+public class RabbitMQEmailAddressContactConfiguration {
+    static final String DEAD_LETTER_EXCHANGE_PREFIX = "TmailQueue-dead-letter-exchange-";
+    static final String DEAD_LETTER_QUEUE_PREFIX = "TmailQueue-dead-letter-queue-";
+    static final String AQMP_URI_PROPERTY = "address.contact.uri";
+    static final String AQMP_MANAGEMENT_URI_PROPERTY = "address.contact.management.uri";
+    static final String AQMP_MANAGEMENT_USER = "address.contact.management.user";
+    static final String AQMP_MANAGEMENT_PASSWORD = "address.contact.management.password";
+    static final String AQMP_EXCHANGE_NAME_PROPERTY = "address.contact.exchange";
+    static final String AQMP_QUEUE_NAME_PROPERTY = "address.contact.queue";
+
+    public static RabbitMQEmailAddressContactConfiguration from(Configuration configuration) {
+        String aqmpURIAsString = configuration.getString(AQMP_URI_PROPERTY);
+        Preconditions.checkState(!Strings.isNullOrEmpty(aqmpURIAsString),
+            String.format("You need to specify the URI of RabbitMQ by '%s' property", AQMP_URI_PROPERTY));
+        URI aqmpUri = URI.create(aqmpURIAsString);
+
+        String managementUriAsString = configuration.getString(AQMP_MANAGEMENT_URI_PROPERTY);
+        Preconditions.checkState(!Strings.isNullOrEmpty(managementUriAsString),
+            String.format("You need to specify the management URI of RabbitMQ by '%s' property", AQMP_MANAGEMENT_URI_PROPERTY));
+        URI managementUri = URI.create(managementUriAsString);
+
+        String managementUser = configuration.getString(AQMP_MANAGEMENT_USER);
+        Preconditions.checkState(!Strings.isNullOrEmpty(managementUser),
+            String.format("You need to specify the %s property as username of rabbitmq management admin account", AQMP_MANAGEMENT_USER));
+        String managementPassword = configuration.getString(AQMP_MANAGEMENT_PASSWORD);
+        Preconditions.checkState(!Strings.isNullOrEmpty(managementPassword),
+            String.format("You need to specify the %s property as password of rabbitmq management admin account", AQMP_MANAGEMENT_PASSWORD));
+        RabbitMQConfiguration.ManagementCredentials managementCredential = new RabbitMQConfiguration.ManagementCredentials(managementUser, managementPassword.toCharArray());
```

Comment:
Now that I'm rereading this again... Didn't we say the rabbitmq could be external, but it could also be the same one used by TMail for the rest? 

I think `address.contact.uri`, `address.contact.management.uri`, `address.contact.management.user` and `address.contact.management.password` should be optional, and defaulting to the default rabbitmq instance if missing?

Reply:
make sense (y)

Reply:
I'm not 100% sure either though...

Reply:
Also we could think of having installs of TMail where there is no external contact app using this... So it should not be mandatory

Reply:
Should we have then also a enable true/false field for this feature?

Reply:
I tried to make it optional, but realize the config looks like complexity. (hard for administrator)
Example:
If `address.contact.uri` has been declare. => we use 3rd rabbitmq.
then `address.contact.management.uri` is must 3rd too, or empty. It can not be the same with tmail rabbitmq

Reply:
wait to me make a commit

Reply:
In fact should completely drop the management uri... So you only have the `address.contact.uri` left. 

How is that hard though? if it's declared you use it, if not you use the one declared in `uri` field? Or is there something tricky?

Reply:
Ah wait.. also we discussed with @chibenwa it's acceptable to put twice the same uri then. Forget what i said if it makes it easier